src/btree/bt_vrfy.c

```diff
@@ -782,11 +782,12 @@ __verify_key_hs(
     wt_timestamp_t older_start_ts, older_stop_ts;
     uint64_t hs_counter;
     uint32_t hs_btree_id;
-    int cmp, exact;
     char ts_string[2][WT_TS_INT_STRING_SIZE];
 
     btree = S2BT(session);
     hs_btree_id = btree->id;
+    WT_RET(__wt_curhs_open(session, NULL, &hs_cursor));
```

Comment:
Do you not need to close this in this function?

Reply:
Nice catch - yes we need to close the cursor. Fixed.